File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/services/HouseImageService.java
package com.chaincuerealestate.chaincuerealestate.services;

import com.chaincuerealestate.chaincuerealestate.domains.HouseImage;
import com.chaincuerealestate.chaincuerealestate.exceptions.HouseImageNotFoundException;
import com.chaincuerealestate.chaincuerealestate.repositories.HouseImageRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class HouseImageService implements HouseImageServiceI {
    private final HouseImageRepository houseImageRepository;

    @Override
    public HouseImage save(String url) {
        var houseImage = HouseImage.create(url);
        return houseImageRepository.save(houseImage);
    }

    @Override
    public HouseImage findById(String id) {
        return houseImageRepository.findById(id).orElseThrow(() -> new HouseImageNotFoundException(id));
    }

    @Override
    public List<HouseImage> findAll() {
        return houseImageRepository.findAll();
    }

}

File: kotlin/chaincue-real-estate/src/main/kotlin/com/example/chaincuerealestate/services/HouseImageService.kt
package com.example.chaincuerealestate.services

import com.example.chaincuerealestate.domains.HouseImage
import com.example.chaincuerealestate.exceptions.HouseImageNotFoundException
import com.example.chaincuerealestate.repositories.HouseImageRepository
import org.springframework.stereotype.Service

@Service
class HouseImageService(private val houseImageRepository: HouseImageRepository) : HouseImageServiceI {

    override fun save(url: String): HouseImage {
        val houseImage = HouseImage.create(url)
        return houseImageRepository.save(houseImage)
    }

    override fun findById(id: String): HouseImage {
        return houseImageRepository.findById(id)
            .orElseThrow { HouseImageNotFoundException(id) }
    }

    override fun findAll(): List<HouseImage> {
        return houseImageRepository.findAll()
    }
}
